app/src/org/commcare/android/database/user/UserDatabaseUpgrader.java

```diff
@@ -208,15 +221,71 @@ private boolean upgradeSevenEight(SQLiteDatabase db, int oldVersion, int newVers
         }
     }
 
-    /**
+    /*
      * Deserialize user fixtures in db using old form instance serialization
      * scheme, and re-serialize them using the new scheme that preserves
      * attributes.
      */
-    private boolean upgradeEightNine(SQLiteDatabase db, int oldVersion, int newVersion) {
+    private boolean upgradeEightNine(SQLiteDatabase db) {
         return FixtureSerializationMigration.migrateFixtureDbBytes(db, c);
     }
 
+    /**
+     * Adding an appId field to FormRecords, for compatibility with multiple apps functionality
+     */
+    private boolean upgradeNineTen(SQLiteDatabase db) {
+        db.beginTransaction();
+        try {
+            if (multipleInstalledAppRecords()) {
+                // Cannot migrate FormRecords once this device has already started installing
+                // multiple applications, because there is no way to know which of those apps the
+                // existing FormRecords belong to
+                throw new MigrationException(true);
```

Comment:
er, do we have a plan for dealing with this? If the app can't start up and we're going to lose these forms anyway, it seems like we don't gain much from either picking a random app and assigning all the forms to it or something?